### ISSUE TITLE
Running runIde in parallel can fail based on a race-condition

### ISSUE THREAD

**Issue by m273d15** [saros-project/saros#724]
Our current solution in order to run the task runIde of the gradle-intellij-plugin multiple times simultaneously is to create a lock file in the used config directory of the running intellij instances.
If a new starting runIde task finds only config dirs with existing lock files a new config dir is created (each config dir has a pattern like `<config base name><config dir number>`).
If the runIde instance is closed via the ui, the lock file is removed and the next runIde instance could use this config directory.

The check whether a directory is locked and the creation of a lock is not atomic and allows a scenario like:
* call runIde (A)
* call runIde (B)
* A: checks for a free config dir C1 and determines that this dir is free
* B: checks for a free config dir C1 and determines that this dir is free and creates the lock
* A: tries to create a lock and fails

This scenario leads to an exception.
It is not clear whether it is worth to fix this issue.

**Comment by m273d15** [comment]
@tobous I would adjust the exception message  (explaining the reason and how to avoid) and close the issue with "wontfix".
Any objections?

**Comment by tobous** [comment]
> @tobous I would adjust the exception message (explaining the reason and how to avoid) and close the issue with "wontfix".
> Any objections?

No, sounds good to me.

**Comment by m273d15** [comment]
Close with "wontfix", because the solution is complex and the benefit is low.
Feel free to re-open if necessary.
